app/component/IndexPage.js

```diff
@@ -105,6 +110,18 @@ class IndexPage extends React.Component {
       this.context.executeAction(storeDestination, destination);
     }
 
+    if (this.context.config.startSearchFromUserLocation) {
+      checkPositioningPermission().then(permission => {
+        if (
+          permission.state === 'granted' &&
+          this.props.locationState.hasLocation === false &&
+          !this.props.locationState.isLocationingInProgress &&
+          this.props.locationState.status === 'no-location'
```

Comment:
Complex if chains are sort of unpleasant. Would it be enough to check state === 'granted' && status === 'no-location' ?